src/coreclr/jit/lowerxarch.cpp

```diff
@@ -5227,34 +5242,101 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* containingNode, Ge
                 case NI_AVX2_ConvertToVector256Int32:
                 case NI_AVX2_ConvertToVector256Int64:
                 {
-                    supportsGeneralLoads = (!node->OperIsHWIntrinsic());
+                    assert(!supportsSIMDScalarLoads);
+
+                    if (!containingNode->OperIsMemoryLoad())
+                    {
+                        // The containable form is the one that takes a SIMD value, that may be in memory.
+
+                        if (!comp->canUseVexEncoding())
+                        {
+                            supportsAlignedSIMDLoads   = true;
+                            supportsUnalignedSIMDLoads = !supportsAlignedSIMDLoads;
+                        }
+                        else
+                        {
+                            supportsAlignedSIMDLoads   = !comp->opts.MinOpts();
```

Comment:
Is the intent here to enabled aligned SIMD loads for optimized code?
```
supportsAlignedSIMDLoads = comp->opts.OptimizationEnabled();
```

Reply:
Yes, it can only be enabled for optimized code and matches what we are already doing for other aligned only folds.

If we were to update the logic, we should do it in a separate PR and ensure they stay consistent.